### PR TITLE
HW14

### PR DESCRIPTION
**Pull request by albina-pasichnyk**
In scope of this pull request functional tests for Human class were implemented.

### CODE REVIEW

File: HW_14/test_human.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    """Test age_limit is private"""
+    human = create_human
+    with pytest.raises(AttributeError, match=f"'{human.__class__.__name__}' object has no attribute '__age_limit'"):
+        assert '100' == human.__age_limit
+
+
+def test_age_property(create_human):
+    """Test age property"""
+    human = create_human
+    assert 18 == human.age
```
Comment (bohdandovbsyh) [inline HW_14/test_human.py]: why 18? where do you get this magic number?

File: HW_14/test_human.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+
+def test_gender_property(create_human):
+    """Test gender property"""
+    human = create_human
+    assert 'male' == human.gender
```
Comment (bohdandovbsyh) [inline HW_14/test_human.py]: why male?

File: HW_14/test_human.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    old_human = create_diff_human(name, 100, 'female')
+    old_human.grow()
+    with pytest.raises(Exception, match=f'{name} is already dead...'):
+        old_human.grow()
+
+
+def test_change_valid_gender(create_human):
+    """Test valid change of human's gender"""
+    male_human = create_human
+    new_gender = 'female'
```
Comment (bohdandovbsyh) [inline HW_14/test_human.py:66]: How do you know that new_gender should be 'female'

File: HW_14/test_human.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    human = create_human
+    assert 'male' == human.gender
+
+
+def test_human_grow(create_human):
+    """Test valid grow of a human"""
+    human = create_human
+    new_age = human.age + 1
+    human.grow()
+    assert new_age == human.age
```
Comment (bohdandovbsyh) [inline HW_14/test_human.py]: please add assertion messages in all asserts

File: HW_14/test_human.py (hunk 1 above)
Comment (albina-pasichnyk) [reply HW_14/test_human.py]: 18 that is the age value used in fixture "create_human". 
That is why it was used as a specific value for assert. 
Should it be better created as separate variable inside the test, in order to make it more explicit?

File: HW_14/test_human.py (hunk 2 above)
Comment (albina-pasichnyk) [reply HW_14/test_human.py]: The same as for age - it's value from fixture. Move it explicitely to test?

**Review by bohdandovbsyh — APPROVED** [review]

**Review by bohdandovbsyh — APPROVED** [review]

**Pull request merged**